verify/verify.go

```diff
@@ -866,20 +866,52 @@ func verifyHash256(quote *pb.QuoteV4) error {
 	return nil
 }
 
-func isSvnHigherOrEqual(svn []byte, components []pcs.TcbComponent) bool {
-	for i := range svn {
-		if svn[i] < components[i].Svn {
+func isCPUSvnHigherOrEqual(pckCertCPUSvnComponents []byte, sgxTcbcomponents []pcs.TcbComponent) bool {
+	for i := range pckCertCPUSvnComponents {
+		if pckCertCPUSvnComponents[i] < sgxTcbcomponents[i].Svn {
 			return false
 		}
 	}
 	return true
 }
 
+func isTdxTcbSvnHigherOrEqual(teeTcbSvn []byte, tdxTcbcomponents []pcs.TcbComponent) bool {
+	start := 0
+	if teeTcbSvn[0] > 0 {
+		start = 2
+	}
+	for i := start; i < len(teeTcbSvn); i++ {
+		if teeTcbSvn[i] < tdxTcbcomponents[i].Svn {
+			return false
+		}
+	}
+	return true
+}
+
+func getMatchingTdxModuleTcbLevel(tcbInfoTdxModuleIdentities []pcs.TdxModuleIdentity, teeTcbSvn []byte) (*pcs.TcbLevel, error) {
+	tdxModuleVersion := []byte(teeTcbSvn[1:2])
+	tdxModuleIdentityID := "TDX_" + hex.EncodeToString(tdxModuleVersion)
+	tdxModuleIsvSvn := uint32(teeTcbSvn[0])
+
+	for _, tdxModuleIdentity := range tcbInfoTdxModuleIdentities {
+		if tdxModuleIdentityID == tdxModuleIdentity.ID {
+			for _, tcbLevel := range tdxModuleIdentity.TcbLevels {
+				if tdxModuleIsvSvn >= tcbLevel.Tcb.Isvsvn {
+					logger.V(2).Info("Matched to TDX Module Identity TCB Level with ISVSVN: ", tcbLevel.Tcb.Isvsvn)
+					return &tcbLevel, nil
+				}
+			}
+			return nil, fmt.Errorf("missing matching TDX Module Identity TCB Level (ISVSVN: %d)", tdxModuleIsvSvn)
+		}
+	}
+	return nil, fmt.Errorf("missing matching TDX Module Identity (%q) for given TEE TDX version (%q)", tdxModuleIdentityID, tdxModuleVersion)
```

Comment:
"could not find a TDX Module Identity (%q) matching the given TEE TDX version (%q)"